### DISCUSSION TITLE
Add per-calendar user agent

### DISCUSSION THREAD

**Discussion by CrossRoast** [bitfireAT/icsx5#249] [Ideas]
It'd be great to add the ability to specify a custom user-agent for CalDAV subscriptions. E.g. https://www.bls.gov/schedule/news_release/bls.ics works on Apple Calendar and in browsers, but throws 403 with curl or icsx5. When I set curl's user-agent to match firefox, it worked fine.

**Comment by sunkup** [comment]
That'd be a very specific addition, of which I can not see great usage of, aside from circumventing this server side protection. The U.S. Bureau of Labor Statistics seems to have its reason for disallowing automated access. 

> The BLS is committed to providing data promptly and according to established schedules. Automated retrieval programs (commonly called "robots" or "bots") can cause delays and interfere with other customers' timely access to information. Therefore, bot activity that doesn't conform to BLS usage policy is prohibited.

Some kind of bot activity, which conforms with usage policy, may be allowed. I am not sure ICSx5 falls under that.

**Reply by CrossRoast** [reply]
I'm sure BLS's intention is to allow people to access its calendar, I've been using it on my iPhone for a while. Someone probably added user-agent checks thinking "what else can we do?". icsx5 is a calendar app, not a scraper.
The addition could be part of a broader patch to add headers control for the user, if you consider this too specific. That would allow support for more authentication methods.
